### DISCUSSION TITLE
What can be done with `..<var>` expressions?

### DISCUSSION THREAD

**Discussion by jimlloyd** [EvgSkv/logica#171] [Q&A] [answered]
The `..<var>` expression seems like it could be very useful when writing predicates that operate on tables with "many" columns. But from my understanding so far I am suspecting that they might flexible enough for one use case I have.

Suppose we have a table that contains one column of high-precision timestamps and a dozen or more columns of FLOAT64, each column being time-series data. The columns are formed from an outer join of several similar table such that for any given high precision timestamp only a subset of the columns have non-null data values. 

We want to transform that table by quantizing the timestamps and doing some kind of aggregation in each column over all of the non-null values in a given quantized time bucket.

Does the `..<var>` notation make it possible to express that transformation without using the actual column names?

**Comment by EvgSkv** [comment, marked as answer]
Unfortunately this functionality currently doesn't work. Logica compiles to SQL and I don't think BigQuery has constructs for this.
In the long term extending the language to allow this makes total sense to me.

If I have a similar task I use some other tools/hacks to turn the table into having a row for each cell of the matrix, turning name of the column into a field. Then apply Logica to result. I understand that this is far from ideal.
